x/programs/runtime/runtime.go

```diff
@@ -82,13 +87,30 @@ func (r *WasmRuntime) Initialize(ctx context.Context, programBytes []byte) (err
 	}
 
 	link := Link{wasmtime.NewLinker(r.store.Engine)}
+
+	// enable wasi logging support only in testing mode
+	if r.cfg.testingOnlyMode {
+		wasiConfig := wasmtime.NewWasiConfig()
+		wasiConfig.InheritStderr()
+		wasiConfig.InheritStdout()
+		r.store.SetWasi(wasiConfig)
+		err = link.DefineWasi()
+		if err != nil {
+			return err
+		}
+	}
+
 	// setup metering
 	r.meter = NewMeter(r.store)
 	_, err = r.meter.AddUnits(r.cfg.meterMaxUnits)
 	if err != nil {
 		return err
 	}
 
+	// reset the config max units to 0 this ensures that any subsequent usage of
+	// this config will not inherit the meter units. ex (program -> program)
+	r.cfg.meterMaxUnits = 0
```

Comment:
This must've been a fun bug.

Reply:
Should it instead be set to the remaining units? Or is that handled elsewhere?

Reply:
Good question, the idea is the caller sets the maxfee for the entire execution. once the parent runtime sets the meter I want this cfg value to be 0. This way if we use the config again for program to program the expectation is the meter should start at 0. We then can handle remaining units with `TransferUnits`. You can find an example of this in the program import example[1].

 https://github.com/ava-labs/hypersdk/blob/9c184578d8d1536964abe34055754e15bde644fe/x/programs/runtime/meter.go#L55
 
 [1] https://github.com/ava-labs/hypersdk/blob/9c184578d8d1536964abe34055754e15bde644fe/x/programs/examples/imports/program/program.go#L170

Reply:
Ah ok. We should probably document this very explicitly (maybe right there?)

Reply:
Made a few other changes here. I think its important that there is only one flow for populating the meter with a single config to reduce the scope of leaky metering.

- new cfg populated by genesis sets `maxFee`
- any other inline runtimes must `TransferUnits` meter <-> meter.

Reply:
I thought this over and the Unit fee really should not be part of the cfg for the reasons I am trying to escape :). I have been back and forth on this but I think it should be a param to Intialize.

Reply:
Ok sounds good.